src/coreComponents/dataRepository/xmlWrapper.cpp

```diff
@@ -71,58 +70,78 @@ template void xmlWrapper::stringToInputVariable( Tensor< real64, 6 > & target, s
+  for( xmlNode const includedNode : targetNode.children( includedListTag ) )
   {
-    // Get the child tag and name
-    string childName = childNode.name();
-    GEOSX_ERROR_IF( childName!="File", "Child nodes of \"Included\" should be named \"File\" " );
-
-    string filePathName = childNode.attribute( "name" ).value();
-    if( filePathName[0] != '/' )
+    for( xmlNode const fileNode : includedNode.children() )
     {
-      filePathName = path + filePathName;
-    }
-    xmlDocument includedXmlDocument;
-    xmlResult result;
-    result = includedXmlDocument.load_file( filePathName.c_str());
-    GEOSX_ERROR_IF( !result, "Attempt to include file ("<<filePathName.c_str()<<") failed\n" );
-
-    // To validate correctly, included files should contain the root Problem node
-    xmlNode includedRootNode = includedXmlDocument.child( "Problem" );
-    for( xmlNode importNode=includedRootNode.first_child(); importNode; importNode=importNode.next_sibling())
-    {
-      targetNode.append_copy( importNode );
+      // Extract the file name and construct full includedDirPath
+      string const includedFilePath = [&]()
+      {
+        GEOSX_THROW_IF_NE_MSG( string( fileNode.name() ), includedFileTag,
+                               GEOSX_FMT( "Child nodes of <{}> should be named <{}>", includedListTag, includedFileTag ),
+                               InputError );
+        xmlAttribute const nameAttr = fileNode.attribute( "name" );
+        GEOSX_THROW_IF( !nameAttr, GEOSX_FMT( "<{}> nodes must have a 'name' attribute", includedFileTag ), InputError );
+        string const fileName = nameAttr.value();
+        return isAbsolutePath( fileName ) ? fileName : joinPath( splitPath( currentFilePath ).first, fileName );
+      }();
+
+      xmlDocument includedXmlDocument;
+      xmlResult const result = includedXmlDocument.load_file( includedFilePath.c_str() );
+      GEOSX_THROW_IF( !result, GEOSX_FMT( "Errors found while parsing included XML file {}\nDescription: {}\nOffset: {}",
+                                          includedFilePath, result.description(), result.offset ), InputError );
+
+      // All included files must contain a root node that must match the target node.
+      // Currently, schema only allows <Included> tags at the top level (inside <Problem>).
+      // We then proceed to merge each nested node from included file with the one in main.
+
+      xmlNode includedRootNode = includedXmlDocument.first_child();
+      GEOSX_THROW_IF_NE_MSG( string( includedRootNode.name() ), string( targetNode.name() ),
+                             "Included document root does not match the including XML node", InputError );
+
+      // Process potential includes in the included file to allow nesting
+      includedXmlDocument.append_child( filePathString ).append_attribute( filePathString ).set_value( includedFilePath.c_str() );
+      addIncludedXML( includedRootNode );
```

Comment:
This is the essential change in this file (and PR). Everything else is just a touch up